kedro/runner/runner.py

```diff
@@ -399,7 +414,20 @@ def _run_node_sequential(
         node, catalog, inputs, is_async, hook_manager, session_id=session_id
     )
 
-    for name, data in outputs.items():
+    items: Iterable = outputs.items()
+    # if all outputs are iterators, then the node is a generator node
+    if all(isinstance(d, Iterator) for d in outputs.values()):
+        # make sure we extract the keys and the chunk streams in the same order
+        # [a, b, c]
+        keys = list(outputs.keys())
+        # [Iterator[chunk_a], Iterator[chunk_b], Iterator[chunk_c]]
+        streams = [outputs[k] for k in outputs]
```

Comment:
I think you are right, we can just replace it with `list(outputs.values())`, since in Python dictionaries are already ordered, so there's no chance we will have a mismatch between `keys` and `streams`. Will change it accordingly.